Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -271,6 +289,14 @@ - (void)validatePathSegment:(NSString *)segment {
   }
 }
 
+- (const FieldPath *)path {
+  return _path.get();
+}
+
+- (std::vector<FieldPath> *)fieldMask {
```

Comment:
Rather than a non-const pointer, could this be a const reference?

Reply:
No, actually there is somewhere that calls and changes fieldMask. I'm on the fence whether to use smart point to make this more clear that somewhere is going to mutate this but not sure since that place is one-time use and not actually take the ownership persistently beyond the callee.